modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3686,16 +3686,7 @@ def _compute_new_widths():
                 new_index = self.index.append([other.index for other in others])
             new_columns = joined_index
             frames = [self] + others
-            if all(frame.has_materialized_dtypes for frame in frames):
-                all_dtypes = [frame.dtypes for frame in frames]
-                if not all(dtypes.empty for dtypes in all_dtypes):
-                    new_dtypes = pandas.concat(all_dtypes, axis=1)
-                    # 'nan' value will be placed in a row if a column doesn't exist in all frames;
-                    # this value is np.float64 type so we need an explicit conversion
-                    new_dtypes.fillna(np.dtype("float64"), inplace=True)
-                    new_dtypes = new_dtypes.apply(
-                        lambda row: find_common_type(row.values), axis=1
-                    )
+            new_dtypes = ModinDtypes.concat([frame._dtypes for frame in frames], axis=0)
```

Comment:
* Could there be a problem with duplicate columns?
* Why was there concatenation along axis 0 before, but now along axis 1?

Reply:
Indeed, but

> So if we're in pandas.concat(axis=0) we should call ModinDtypes.concat(axis=0) as well, and vice versa with axis=1.

There was `new_dtypes = pandas.concat(all_dtypes, axis=1)`. Should we use `ModinDtypes.concat(axis=1)`? Didn't get it.

Reply:
indeed, on the second thought it seems logical to switch axes